src/v0/destinations/mp/deleteUsers.js

```diff
@@ -62,12 +110,31 @@ const userDeletionHandler = async (userAttributes, config) => {
       }
     }),
   );
+
   return {
     statusCode: 200,
     status: 'successful',
   };
 };
 
+/**
+ * This function will help to delete the users one by one from the userAttributes array.
+ * @param {*} userAttributes Array of objects with userId, email and phone
+ * @param {*} config Destination.Config provided in dashboard
+ * @returns
+ */
+const userDeletionHandler = async (userAttributes, config) => {
+  if (!config?.token) {
+    throw new ConfigurationError('API Token is a required field for user deletion');
+  }
+
+  if (config?.userDeletionApi === 'task') {
+    return createDeletionTask(userAttributes, config);
+  }
+
+  return deleteProfile(userAttributes, config);
+};
+
 const processDeleteUsers = (event) => {
```

Comment:
Should there be an `async` to this function ?
And also we might need to `await` on `userDeletionHandler` right ?

Reply:
Yes, async/ await is needed here. It is missing in some of the deleteUser implementations. I will address those in a separate PR. In this PR i have added the async/await for mixpanel

Reply:
Added for other destinations in [this](https://github.com/rudderlabs/rudder-transformer/pull/2183) PR.

Reply:
🆒